packages/react-native-windows-init/src/Cli.ts

```diff
@@ -335,34 +342,56 @@ function installReactNativeWindows(
  * @param value The unsanitized value of the option.
  * @returns The sanitized value of the option.
  */
-function optionSanitizer(key: string, value: any): any {
+function optionSanitizer(key: keyof WindowsInitOptions, value: any): any {
+  // Do not add a default case here.
+  // Strings risking PII should just return true if present, false otherwise.
+  // All others should return the value (or false if undefined).
   switch (key) {
     case 'namespace':
-      return value === undefined ? false : true;
-    default:
+    case 'nuGetTestFeed':
+    case 'nuGetTestVersion':
+      return value ? true : false;
+    case 'verbose':
+    case 'version':
+    case 'telemetry':
+    case 'language':
+    case 'overwrite':
+    case 'projectType':
+    case 'experimentalNuGetDependency':
+    case 'useHermes':
+    case 'useWinUI3':
+    case 'useDevMode':
       return value === undefined ? false : value;
   }
 }
 
 /**
  * Sets up and starts the telemetry gathering for the CLI command.
+ * @param args The raw CLI args.
+ * @param options The CLI args parsed by yargs.
  */
-async function startTelemetrySession() {
-  if (!argv.telemetry) {
-    if (argv.verbose) {
+async function startTelemetrySession(
+  args: string[],
+  options: YargsOptionsType,
+) {
+  if (!options.telemetry) {
+    if (options.verbose) {
       console.log('Telemetry is disabled');
     }
     return;
   }
 
   await Telemetry.setup();
 
-  const sanitizedOptions = yargsOptionsToOptions(argv, optionSanitizer);
+  const sanitizedOptions = yargsOptionsToOptions(
+    options,
+    optionSanitizer as OptionSanitizer,
+  );
   const sanitizedDefaultOptions = yargsOptionsToOptions(
     yargsParser.parse(''),
-    optionSanitizer,
+    optionSanitizer as OptionSanitizer,
```

Comment:
Why the needed cast here and above?

Reply:
Changing to the keyof for the key param breaks this for some reason.